src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -32,6 +33,16 @@ private void CheckDisposedOrStarted()
             }
         }
 
+        public Func<string, int, CancellationToken, ValueTask<Stream>> ConnectCallback
+        {
+            get => _settings._customConnect;
+            set
+            {
+                CheckDisposedOrStarted();
+                _settings._customConnect = value;
```

Comment:
What does it mean to allow null to be set here?  Won't we null ref later when we try to invoke this?